ShellCheck fixes to pass shell linter checks

mpepping opened the pull request:
**Description:** 

Fixes shellcheck suggestions in `init-eks-fargate-cluster-receiver.sh` to pass shell linter checks.

**Link to Splunk idea:** n/a
**Testing:**  Run the shellscript 

**Documentation:** n/a

mpepping commented:
I have read the CLA Document and I hereby sign the CLA

jvoravong commented:
@mpepping 
- Tt looks like this change [breaks](https://github.com/signalfx/splunk-otel-collector-chart/actions/runs/7724500269/job/21084496422?pr=1133) our pre-commit CI/CD workflow.
- We should also test this change works in an EKS fargate cluster properly.
Can you look into these?

mpepping commented:
@jvoravong, Updated the test render accordingly via `make render`.  Also reverted the `yq` change in favour of a shellcheck exception. Pre-commit and shell lint both pass.

jvoravong commented:
I closed and reopened the pull request so the test would rerun with the "Skip Changelog" label applied.

jvoravong commented:
@mpepping thanks for your contribution.
